### PR TITLE
Migrate to Google Kubernetes Engine

### PR DESCRIPTION
**Pull request by thebeanogamer**
As a result of @swedg's descision to no longer host the bot, it will be migrating to Google Kubernetes Engine.

The majority of the required changes have been made, and this PR will complete the migration.

It also includes a few QOL fixes for Azure Devops which weren't large enough to need their own PRs.

### CODE REVIEW

**Comment by Ge0rg3** [comment]
Please reprogram in Scratch or Visual Basic.

File: cdbot/__init__.py [hunk 1]
```diff
@@ -10,5 +11,5 @@
 def main():
     """Entry point for poetry script."""
     if environ.get("SENTRY_URL") is not None:
-        sentry_sdk.init(environ.get("SENTRY_URL"))
-    bot.run(environ.get("BOT_TOKEN"))
+        sentry_sdk.init(base64.b64decode(environ.get("SENTRY_URL")))
```
Comment (jos-b) [inline cdbot/__init__.py]: This is not nice, could this be stored in the constants file so you can pass it straight into the function?

File: cdbot/__init__.py [hunk 2]
```diff
@@ -10,5 +11,5 @@
...
+    bot.run(base64.b64decode(environ.get("BOT_TOKEN")))
```
Comment (jos-b) [inline cdbot/__init__.py]: This is not nice, could this be stored in the constants file so you can pass it straight into the function?

File: cdbot/cogs/fun.py [hunk 3]
```diff
@@ -99,7 +101,12 @@ def __init__(self, bot: Bot):
         if message.channel.id == QUOTES_CHANNEL_ID and (
             message.author.id == QUOTES_BOT_ID or message.mentions is not None
         ):
-            conn = await asyncpg.connect()
+            conn = await asyncpg.connect(host=base64.b64decode(environ.get("PGHOST")),
```
Comment (jos-b) [inline cdbot/cogs/fun.py]: This is not nice, could this be stored in the constants file so you can pass it straight into the function?

File: cdbot/cogs/fun.py [hunk 4]
```diff
@@ -99,7 +101,12 @@ def __init__(self, bot: Bot):
...
+                                         port=base64.b64decode(environ.get("PGPORT")),
```
Comment (jos-b) [inline cdbot/cogs/fun.py]: This is not nice, could this be stored in the constants file so you can pass it straight into the function?

File: cdbot/cogs/fun.py [hunk 5]
```diff
@@ -99,7 +101,12 @@ def __init__(self, bot: Bot):
...
+                                         user=base64.b64decode(environ.get("PGUSER")),
```
Comment (jos-b) [inline cdbot/cogs/fun.py]: This is not nice, could this be stored in the constants file so you can pass it straight into the function?

File: cdbot/cogs/fun.py [hunk 6]
```diff
@@ -99,7 +101,12 @@ def __init__(self, bot: Bot):
...
+                                         password=base64.b64decode(environ.get("PGPASSWORD")),
```
Comment (jos-b) [inline cdbot/cogs/fun.py]: This is not nice, could this be stored in the constants file so you can pass it straight into the function?

File: cdbot/cogs/fun.py [hunk 7]
```diff
@@ -99,7 +101,12 @@ def __init__(self, bot: Bot):
...
+                                         database=base64.b64decode(environ.get("PGDATABASE")))
```
Comment (jos-b) [inline cdbot/cogs/fun.py]: This is not nice, could this be stored in the constants file so you can pass it straight into the function?

File: cdbot/cogs/fun.py [hunk 8]
```diff
@@ -253,7 +260,11 @@ def check(reaction, user):
         Returns a random quotation from the #quotes channel.
         A user can be specified to return a random quotation from that user.
         """
-        conn = await asyncpg.connect()
+        conn = await asyncpg.connect(host=base64.b64decode(environ.get("PGHOST")),
```
Comment (jos-b) [inline cdbot/cogs/fun.py]: This is not nice, could this be stored in the constants file so you can pass it straight into the function?

File: cdbot/cogs/fun.py [hunk 9]
```diff
@@ -253,7 +260,11 @@ def check(reaction, user):
...
+                                     port=base64.b64decode(environ.get("PGPORT")),
```
Comment (jos-b) [inline cdbot/cogs/fun.py]: This is not nice, could this be stored in the constants file so you can pass it straight into the function?

File: cdbot/cogs/fun.py [hunk 10]
```diff
@@ -253,7 +260,11 @@ def check(reaction, user):
...
+                                     user=base64.b64decode(environ.get("PGUSER")),
```
Comment (jos-b) [inline cdbot/cogs/fun.py]: This is not nice, could this be stored in the constants file so you can pass it straight into the function?

File: cdbot/cogs/fun.py [hunk 11]
```diff
@@ -253,7 +260,11 @@ def check(reaction, user):
...
+                                     password=base64.b64decode(environ.get("PGPASSWORD")),
```
Comment (jos-b) [inline cdbot/cogs/fun.py]: This is not nice, could this be stored in the constants file so you can pass it straight into the function?

File: cdbot/cogs/fun.py [hunk 12]
```diff
@@ -253,7 +260,11 @@ def check(reaction, user):
...
+                                     database=base64.b64decode(environ.get("PGDATABASE")))
```
Comment (jos-b) [inline cdbot/cogs/fun.py]: This is not nice, could this be stored in the constants file so you can pass it straight into the function?

File: cdbot/cogs/fun.py [hunk 13]
```diff
@@ -319,7 +330,11 @@ def check(reaction, user):
         Pulls all quotes from a quotes channel into a PostgreSQL database.
         Needs PGHOST, PGPORT, PGUSER, PGDATABASE and PGPASSWORD env vars.
         """
-        conn = await asyncpg.connect()
+        conn = await asyncpg.connect(host=base64.b64decode(environ.get("PGHOST")),
```
Comment (jos-b) [inline cdbot/cogs/fun.py]: This is not nice, could this be stored in the constants file so you can pass it straight into the function?

File: cdbot/cogs/fun.py [hunk 14]
```diff
@@ -319,7 +330,11 @@ def check(reaction, user):
...
+                                     port=base64.b64decode(environ.get("PGPORT")),
```
Comment (jos-b) [inline cdbot/cogs/fun.py]: This is not nice, could this be stored in the constants file so you can pass it straight into the function?

File: cdbot/cogs/fun.py [hunk 15]
```diff
@@ -319,7 +330,11 @@ def check(reaction, user):
...
+                                     user=base64.b64decode(environ.get("PGUSER")),
```
Comment (jos-b) [inline cdbot/cogs/fun.py]: This is not nice, could this be stored in the constants file so you can pass it straight into the function?

File: cdbot/cogs/fun.py [hunk 16]
```diff
@@ -319,7 +330,11 @@ def check(reaction, user):
...
+                                     password=base64.b64decode(environ.get("PGPASSWORD")),
```
Comment (jos-b) [inline cdbot/cogs/fun.py]: This is not nice, could this be stored in the constants file so you can pass it straight into the function?

File: cdbot/cogs/fun.py [hunk 17]
```diff
@@ -319,7 +330,11 @@ def check(reaction, user):
...
+                                     database=base64.b64decode(environ.get("PGDATABASE")))
```
Comment (jos-b) [inline cdbot/cogs/fun.py]: This is not nice, could this be stored in the constants file so you can pass it straight into the function?

**Comment by thebeanogamer** [comment]
@jos-b These files cannot be stored in constants.py, as they need to be kept secret and added by GKE, however, I will try and make it more efficient.

File: cdbot/cogs/fun.py [hunk 18]
```diff
@@ -15,7 +15,7 @@
 from aiohttp import ClientSession
 from cdbot.constants import (
     ADMIN_ROLES, EMOJI_LETTERS, FAKE_ROLE_ID, QUOTES_BOT_ID, QUOTES_CHANNEL_ID,
-    STAFF_ROLE_ID, WELCOME_BOT_ID
+    STAFF_ROLE_ID, WELCOME_BOT_ID, Postgresql
```
Comment (jos-b) [inline cdbot/cogs/fun.py]: Little nitpick but can we call this `PostgreSQL`, looks nicer than Postgresql imo.

File: cdbot/constants.py [hunk 19]
```diff
@@ -4,7 +4,7 @@
 """
 Setup PostgreSQL
 """
-class Postgresql():
+class PostgreSQL():
```
Comment (jos-b) [inline cdbot/constants.py]: Un-needed parenthesis.

File: cdbot/__init__.py (hunk 2 above)
Comment (thebeanogamer) [reply cdbot/__init__.py]: I've not moved this to constants.py as it is only called once.

File: cdbot/__init__.py (hunk 1 above)
Comment (thebeanogamer) [reply cdbot/__init__.py]: I've not moved this to constants.py as it is only called once.

**Pull request merged**